### DISCUSSION TITLE
getUrl() and getTemporaryUrl()

### DISCUSSION THREAD

**Discussion by CaReS0107** [spatie/laravel-medialibrary#2918] [Q&A]
These two methods return a path to the file which was uploaded without a problem.

The interesting part is if I have a pdf file with the name `some.pdf` uploaded on the server or local storage and then I try to upload another file with the same name but different content. If I get all media lists the path to all files is the same.
And all path is pointing to the latest uploaded file.

As far as I can see media library doesn't have a method to indicate from what column it must take the file name 
`/tests/TestSupport/temp/media/1/some.pdf` it generates a path base on the `file_name` column from the database. It would be great to have a method that can indicate to take file names from another column.

Or it probably have some method to avoid this kind of issue?

**Comment by 4unkur** [comment]
It's not the same. the url will contain media ID, which is not same for all records
